### ISSUE TITLE
Many issues don't get implemented, so we should revisit them

### ISSUE THREAD

**Issue by penguinmoose** [ScratchAddons/ScratchAddons#5749]
Usually, when an issue gets opened, there is a period of discussion when people see the new issue. However, if the issue doesn't get resolved by then, the issue floats down the page, which means fewer people view it. This creates slower and slower progress. Old issues may remain untouched for months.

Yes, there will be a steady flow of new issues, and some will get quickly implemented. However, the ones that don't will simply add to the ever-growing pile of old discussions.

I suggest that we spend time resolving these old issues. We have over 500 open issues, but only a small portion of them are in discussion. The rest are simply "sitting there". If a given issue is relevant, we should discuss it; if it is clear that we can't implement it (or already did), we should close it.

However, it might be challenging to resolve all of these issues, old and new, and still keep up with the flow of new requests without forgetting some. Because of this, the first thing we should do to an issue is to mark its priority. High-priority ones that are relevant, even if old, should be resolved as soon as possible. These include bug reports, much-needed core changes, and important features to implement. Many old issues have unmarked priorities, so it will be hard to find important issues. If we always mark the priority of each issue, we know which ones we need to implement first. 

Another solution is to add all issues to the project board. We can, too, mark its priority and difficulty. Low-priority issues can go into the backlog, and high-priority ones can go into "focus". If we put all issues on the project board, it will be very easy to sort and organize our issues.

**Comment by BroJac5246** [comment]
I definitely think we should try to add priority labels to more issues, and we should all try to be more proactive about looking through old issues. I'm not so sure about adding everything to the project board, though. 547 issues is a lot to have in one place.

**Comment by DNin01** [comment]
I can see that here. It can be hard to deal with, organize, decide on, and plan tasks when you have this many feature requests, no matter what you're working on. I've noticed this too and it makes we wish that everyone in the community could work together to figure out what to do, and I wish I could see what other people are wanting most. This exact thing happens in other projects too - it's natural for suggestions to sink and not get noticed, even with more flexible task organization setups. This sort of stuff even happens in communities and level editor games like Super Mario Maker, where great creations don't get the attention they deserve.

We can put issues on the What's Happening? board, but [GitHub limits us to 1,200 items](https://github.com/orgs/community/discussions/9678). And even then, that could get clogged up.

I'm not sure how much we can do to revive every suggestion, but we can try. Maybe we can make more priority levels, get more people involved in upvoting and stuff, etc.

**Comment by penguinmoose** [comment]
> I definitely think we should try to add priority labels to more issues, and we should all try to be more proactive about looking through old issues. I'm not so sure about adding everything to the project board, though. 547 issues is a lot to have in one place.

That's true. I think adding _high-priority_ issues sounds more reasonable. There might be some workflow to automatically do that for us.

**Comment by penguinmoose** [comment]
@BroJac5246 Also, most issues don't have a priority label, so always attaching a priority to each issue will make it easy to find important ones.

**Comment by BroJac5246** [comment]
> @BroJac5246 Also, most issues don't have a priority label, so always attaching a priority to each issue will make it easy to find important ones.

Yeah, we need to be better about that.

**Comment by DNin01** [comment]
One of my repositories has _six_ priority-level labels.

**Comment by penguinmoose** [comment]
> > @BroJac5246 Also, most issues don't have a priority label, so always attaching a priority to each issue will make it easy to find important ones.
> 
> Yeah, we need to be better about that. Labels are the key tool here.

But what if it's hard to tell if a given issue is important or not? For example, there might be a severe bug, but we might not know if it happens to many people or not. If we assume that a high number of people are affected, we will probably rate that bug high-priority. However, if that issue is actually not that important, spending our limited time and resources on that issue will cause fixes and changes that are more important to be delayed.

Prioritizing things correctly is very important for spending time effectively. However, it is often difficult to tell the exact priority of an issue the moment it's created (and not wait until it is further discussed). But not labeling things quickly is a problem, as discussed in the first post.

**Comment by DNin01** [comment]
> > > @BroJac5246 Also, most issues don't have a priority label, so always attaching a priority to each issue will make it easy to find important ones.
> 
> > 
> 
> > Yeah, we need to be better about that. Labels are the key tool here.
> 
> 
> 
> But what if it's hard to tell if a given issue is important or not?
> 
> 
> 
> For example, there might be a severe bug, but we might not know if it happens to many people or not. If we assume that a high number of people are affected, we will probably rate that bug high-priority. However, if that issue is actually not that important, spending our limited time and resources on that issue will cause fixes and changes that are more important to be delayed.
> 
> 
> 
> Prioritizing things correctly is very important for spending time effectively. However, it is often difficult to tell the exact priority of an issue the moment it's created (and not wait until it is further discussed). But not labeling things quickly is a problem, as discussed in the first post.

It may be that we also need to find a way to get the community involved and favor what the community is interested in. If enough people think of some feature and realize there's an existing issue for it and it gets lots of upvotes, then we can consider it and maybe increase the priority. I don't know how much we're already doing this.

**Comment by penguinmoose** [comment]
@DNin01 I don't think our community is going to be active enough to "instantly" rate new issues.

If we assume the highest priority that a certain issue possibly is, that issue will receive more attention, so we can rate it more accurately faster.

**Comment by DNin01** [comment]
> By "getting the community involved", do you mean doing some kind of discussion about the topic of prioritizing issues better?

Mostly in what features we should add next.

**Comment by penguinmoose** [comment]
> > By "getting the community involved", do you mean doing some kind of discussion about the topic of prioritizing issues better?
> 
> Mostly in what features we should add next.

I see. So instead of only discussing specific issues that get opened, we should have an open discussion about what we should add or change in the future? Like making a to-do list?

**Comment by DNin01** [comment]
> > > By "getting the community involved", do you mean doing some kind of discussion about the topic of prioritizing issues better?
> 
> > 
> 
> > Mostly in what features we should add next.
> 
> 
> 
> I see. So instead of only discussing specific issues that get opened, we should have an open discussion about what we should add or change in the future? Like making a to-do list?

No. Actually, I didn't mean running surveys and stuff, but if more users could be suggesting things and looking at existing suggestions, it might help the ideas get more attention. Also, this isn't counting the Discord server (I don't know what's going on there because I don't use Discord - maybe that's a better place for suggestions and GitHub Issues is a better place for things we're going to do).

We could also open a public community forum of our own so anyone could join in.

**Comment by BroJac5246** [comment]
> > > > By "getting the community involved", do you mean doing some kind of discussion about the topic of prioritizing issues better?
> > 
> > > 
> > 
> > > Mostly in what features we should add next.
> > 
> > 
> > 
> > I see. So instead of only discussing specific issues that get opened, we should have an open discussion about what we should add or change in the future? Like making a to-do list?
> 
> No. Actually, I didn't mean running surveys and stuff, but if more users could be suggesting things and looking at existing suggestions, it might help the ideas get more attention. Also, this isn't counting the Discord server (I don't know what's going on there because I don't use Discord - maybe that's a better place for suggestions and GitHub Issues is a better place for things we're going to do).
> 
> We could also open a public community forum of our own so anyone could join in.

I've been thinking about the Discord server as well. It's a great place for outreach; my problem is that, like you, I don't really use Discord. We should consider the benefits of it, though.

**Comment by DNin01** [comment]
There are probably several people who are simply unwilling to create a Discord account just to join our community, or maybe they're unsure about it. If we had something where people could create a Scratch Addons account (ooh and imagine settings backups) or something like that in order to join a community forum of our own (we could even make it comply with laws which would allow people under 13 to join in), I think we'd get quite a lot more people. That might not be financially possible for the team right now, though.

**Comment by WorldLanguages** [comment]
We plan to have a feedback page soon. Most new addon ideas (that don't have any technical discussion) can be moved there, and everyday users (that don't have a GitHub account) could upvote them and discuss them. We already have this, in a way, in the community Discord server.
Ideally GitHub issues should only exist if there's any relevant technical discussion. Writing a new addon idea is way easier than actually developing it, so it shouldn't be surprising that there's so many open issues.
Maybe we should have a system where we keep checking some old issues every day to see if they're still relevant. Sorting by [least recently updated](https://github.com/ScratchAddons/ScratchAddons/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-asc) is the closest we can get right now.

Thanks for taking the time to discuss these things, we should do it more often. I never actually expected Scratch Addons to grow this big, so it's hard to keep up.

**Comment by penguinmoose** [comment]
> > > > By "getting the community involved", do you mean doing some kind of discussion about the topic of prioritizing issues better?
> > 
> > 
> > > 
> > 
> > 
> > > Mostly in what features we should add next.
> > 
> > 
> > I see. So instead of only discussing specific issues that get opened, we should have an open discussion about what we should add or change in the future? Like making a to-do list?
> 
> No. Actually, I didn't mean running surveys and stuff, but if more users could be suggesting things and looking at existing suggestions, it might help the ideas get more attention. Also, this isn't counting the Discord server (I don't know what's going on there because I don't use Discord - maybe that's a better place for suggestions and GitHub Issues is a better place for things we're going to do).
> 
> We could also open a public community forum of our own so anyone could join in.

I see. I agree that getting more people involved in discussing suggestions, etc will definitely help. A public forum is a good idea, but we will need some way to synchronize suggestions between GitHub and that forum.

**Comment by penguinmoose** [comment]
> We plan to have a feedback page soon. Most new addon ideas (that don't have any technical discussion) can be moved there, and everyday users (that don't have a GitHub account) could upvote them and discuss them. We already have this, in a way, in the community Discord server. Ideally GitHub issues should only exist if there's any relevant technical discussion. Writing a new addon idea is way easier than actually developing it, so it shouldn't be surprising that there's so many open issues. Maybe we should have a system where we keep checking some old issues every day to see if they're still relevant. Sorting by [least recently updated](https://github.com/ScratchAddons/ScratchAddons/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-asc) is the closest we can get right now.
> 
> Thanks for taking the time to discuss these things, we should do it more often. I never actually expected Scratch Addons to grow this big, so it's hard to keep up.

I agree that it will be helpful to move high-level ideas to another place for discussion. Ideally, that place does not require some third-party account. For example, some users may not have a discord account.

However, GitHub should still be a place for suggestions that have been broken down into a step-by-step checklist for how to accomplish that. When we have decided to implement a suggestion, we should move it to GitHub for some more specific discussion about how we can implement that suggestion.

By moving ideas that still need high-level discussion to another place, we can reduce the amount of issues on GitHub, which makes it easier to know what to focus on.
